File: MainGame/game/Global/Adapter/AStar2DAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using BackEnd4IdleStrategyFS.Godot;
using Godot;
using ZeromaXPlayground.game.Global.Common;

namespace ZeromaXPlayground.game.Global.Adapter;

public class AStar2DAdapter(AStar2D aStar2D) : IAdapter.IAStar2D
{
    public void AddPoint(int id, Tuple<int, int> t)
    {
        // GD.Print("AStar2D.AddPoint");
        aStar2D.AddPoint(id, BackEndUtil.From(t));
    }

    public void ConnectPoints(int fromId, int toId)
    {
        GD.Print($"AStar2D.ConnectPoints ({fromId}, {toId})");
        aStar2D.ConnectPoints(fromId, toId);
    }

    public IEnumerable<int> GetPointConnections(int id)
    {
        GD.Print($"AStar2D.GetPointConnections {id}");
        var connectNavIdArr = aStar2D.GetPointConnections(id);
        return connectNavIdArr.Select(l => (int)l);
    }
}

File: MainGame/FrontEnd4IdleStrategyFS/Global/Adaptor/AStar2DAdapter.fs
namespace FrontEnd4IdleStrategyFS.Global.Adaptor

open Godot
open BackEnd4IdleStrategyFS.Godot.IAdapter
open FrontEnd4IdleStrategyFS.Global.Common

type AStar2DAdapter(aStar2D: AStar2D) =
    interface IAStar2D with
        member this.AddPoint id coord =
            aStar2D.AddPoint(id, BackEndUtil.from coord)

        member this.ConnectPoints fromId toId =
            // GD.Print $"AStar2D.ConnectPoints ({fromId}, {toId})"
            aStar2D.ConnectPoints(fromId, toId)

        member this.GetPointConnections id =
            // GD.Print $"AStar2D.GetPointConnections {id}"
            let connectNavIdArr = aStar2D.GetPointConnections id
            connectNavIdArr |> Seq.ofArray |> Seq.map int
